packages/xforms-engine/src/client/SelectNode.ts

```diff
@@ -6,14 +7,13 @@ import type { LeafNodeDefinition } from '../parse/model/LeafNodeDefinition.ts';
 import type { BaseValueNode, BaseValueNodeState } from './BaseValueNode.ts';
 import type { NodeAppearances } from './NodeAppearances.ts';
 import type { RootNode } from './RootNode.ts';
-import type { TextRange } from './TextRange.ts';
 import type { ValueType } from './ValueType.ts';
 import type { GeneralParentNode } from './hierarchy.ts';
 import type { LeafNodeValidationState } from './validation.ts';
 
-export interface SelectItem {
-	get label(): TextRange<'item-label'>;
-	get value(): string;
+export interface SelectItem extends BaseItem {
+	geometry?(): string;
+	metadata?: Array<{ label: string; value(): string }>;
```

Comment:
`properties` or `additionalProperties`?

`geometry` could also simply be an `additionalProperty`.

But that makes me think -- in what format is the `geometry` value right now when items are from a geoJSON file? Ideally the engine exposes a consistent format so the client doesn't have to do any conversion.

If `geometry` is made into its own concept, we should use a type to describe/validate/represent the structure rather than a raw string.